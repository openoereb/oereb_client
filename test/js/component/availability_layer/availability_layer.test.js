import VectorLayer from 'ol/layer/Vector';
import React from "react";
import {render} from '@testing-library/react'
import {Provider} from "react-redux";

import OerebAvailabilityLayer from
  "../../../../oereb_client/static/src/component/availability_layer/availability_layer";
import MainStore from "../../../../oereb_client/static/src/store/main";

describe('availability layer component', () => {

  let component;

  beforeEach(() => {
    const layer = new VectorLayer();
    component = render(
      <Provider store={MainStore}>
        <OerebAvailabilityLayer availabilityLayer={layer} />
      </Provider>
    );
  });

  it('should render availability layer element', () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

});