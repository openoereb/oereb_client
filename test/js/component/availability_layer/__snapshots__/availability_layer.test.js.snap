// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`availability layer component should render availability layer element 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <OerebAvailabilityLayer
    availabilityLayer={
      VectorLayer {
        "background_": undefined,
        "className_": "ol-layer",
        "declutter_": false,
        "dispatching_": Object {},
        "disposed": false,
        "eventTarget_": undefined,
        "listeners_": Object {
          "change:source": Array [
            [Function],
          ],
        },
        "mapPrecomposeKey_": null,
        "mapRenderKey_": null,
        "ol_uid": "2",
        "on": [Function],
        "once": [Function],
        "pendingRemovals_": Object {},
        "renderBuffer_": 100,
        "renderer_": null,
        "revision_": 2,
        "sourceChangeKey_": null,
        "state_": null,
        "styleFunction_": [Function],
        "style_": [Function],
        "un": [Function],
        "updateWhileAnimating_": false,
        "updateWhileInteracting_": false,
        "values_": Object {
          "maxResolution": Infinity,
          "maxZoom": Infinity,
          "minResolution": 0,
          "minZoom": -Infinity,
          "opacity": 1,
          "source": null,
          "visible": true,
          "zIndex": undefined,
        },
      }
    }
  />
</Provider>
`;
